api/capi/include/nntrainer-tizen-internal.h

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   nntrainer-tizen-internal.h
+ * @date   30 June 2021
+ * @brief  NNTrainer CAPI header for tizen interanl api.
+ * @note   This header is designed to be used only in Tizen
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#ifndef __NNTRAINER_TIZEN_INTERNAL_H__
+#define __NNTRAINER_TIZEN_INTERNAL_H__
+
+#include <nntrainer.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+/**
+ * @brief Save the model after training
+ * @details Use this function to save the current model. @a format rules
+ * how and which part of the model should be saved. Some formats may save
+ * parameters required for training. Some other formats may save model
+ * configurations. The model must be compiled and has allocated memory before
+ * calling this function.
+ * @see @a ml_train_model_format_e to check which part of the model is
+ * saved
+ * @note This function overrides the existing file without any notification.
+ *
+ * @since_tizen 6.5
+ * @param[in] model The NNTrainer model handle to save
+ * @param[in] file_path File path to save the file.
+ * @param[in] format Format flag to determine which format should be used to
+ * save
+ * @return @c 0 on success, Otherwise a negative error value
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER The given @a file_path is
+ * invalid or @a model is not compiled.
+ */
+int ml_train_model_save(ml_train_model_h model, const char *file_path,
+                        ml_train_model_format_e format);
+
+/**
+ * @brief Load the model
+ * @details Use this function to load the current model. @a format rules
+ * how and which part of the model should be saved. Some formats may save
+ * parameters required for training. Some other formats may save model
+ * configurations. The model must be compiled and has allocated memory before
+ * calling this function.
+ * @see @a ml_train_model_format_e to check which part of the model is
+ * saved
+ *
+ * @since_tizen 6.5
+ * @param[in] model The NNTrainer model handle to load.
+ * @param[in] file_path File path to save the file.
+ * @param[in] format Format flag to determine which format should be used to
+ * save
+ * @return @c 0 on success, Otherwise a negativse error value
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_INVALID_PARAMETER The given @a file_path is
+ * invalid or @a model is not compiled.
+ */
+int ml_train_model_load(ml_train_model_h model, const char *file_path,
+                        ml_train_model_format_e format);
```

Comment:
We usually use different name of ini and bin. Even we have option to set the bin name in ini file. I wonder that it is also considered? ( file_path means ini file path?? )

Reply:
Yes, if we state `ML_TRAIN_MODEL_FORMAT_INI_WITH_BIN`, bin name will be generated based on the ini file.

Reply:
So, in the two cases, with BIN/INI format, the user is expected to give the correct file format?

Reply:
the saving file format will be ini as `INI_WITH_BIN` states. I think there has to be main format which points to another file, in the case of two files

Reply:
in case of ML_TRAIN_MODEL_FORMAT_BIN, it will be .bin right?

Reply:
Yup

Reply:
I think this discussion is leading to https://github.com/nnstreamer/nntrainer/pull/1345#discussion_r661396237. Let's continue it there.